Formula/gnirehtet.rb

```diff
@@ -0,0 +1,36 @@
+class Gnirehtet < Formula
+  desc "Reverse tethering tool for Android"
+  homepage "https://github.com/Genymobile/gnirehtet"
+  url "https://github.com/Genymobile/gnirehtet/releases/download/v2.3/gnirehtet-java-v2.3.zip"
+  sha256 "93d1d46ee566376596f033832626dd5e89e76c91f2c46d2383735937b7d3b8b0"
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "gnirehtet.jar"
+    libexec.install "gnirehtet.apk"
+
+    (bin/"gnirehtet").write <<~EOS
+      #!/bin/bash
+      if [[ "$1" == "install" ]]; then
+        shift
+        echo "Installing #{libexec}/gnirehtet.apk"
+        adb install -r #{libexec}/gnirehtet.apk
+      else
+        java -jar #{libexec}/gnirehtet.jar $*
+      fi
+    EOS
+  end
+
+  def caveats; <<~EOS
+    At runtime, adb must be accessible from your PATH.
+
+    You can install adb from Homebrew Cask:
+      brew cask install android-platform-tools
+  EOS
+  end
+
+  test do
+    system bin/"gnirehtet"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Cool beans. I added a test to make sure we can connect to the tunnel.